### DISCUSSION TITLE
Support OSM by MAUI MapControl

### DISCUSSION THREAD

**Discussion by mattleibow** [CommunityToolkit/Maui#866] [New Feature Discussions]
---
**Issue moved from dotnet/maui#12378**
- Please respond to @AllNickNamesAreBusySoITakenThis.

---

_From @AllNickNamesAreBusySoITakenThis on Monday, January 2, 2023 12:03:03 PM_

Your MapControl looks great! Many thanks for it! But do you plan to integrate Open Street Map (OSM) support?

**Comment by mattleibow** [comment]
---
**Issue moved from dotnet/maui#12378**
- Please respond to @Ghostbird.

---

_From @Ghostbird on Tuesday, January 3, 2023 7:32:57 AM_

What do you mean by OSM support? There is not one OSM tile server URI you can use, as you can do for Google Maps (and presumably Bing/Apple and maps).  I would like to see generic XYZ tiles layer support, but it might require quite a bit of work, depending on the current state of the control.

A first version might just allow a replacement of the current map layer with a generic XYZ tiles layer based on a URI template. But then the projection of those tiles probably must be EPSG:3857

EDIT: I had a quick look at the files, and it seems that Google Maps is quite hard-coded into several classes. Probably an abstraction is missing there, something akin to [OpenLayers' `Source`](https://openlayers.org/en/latest/apidoc/module-ol_source_Source-Source.html).

**Reply by AllNickNamesAreBusySoITakenThis** [reply]
When I wrote about OSM support, I imagine smth like Mapsui Map control has now. There we can create a Tile layer with specified tile source. And we can use OSM for it.

**Comment by mattleibow** [comment]
---
**Issue moved from dotnet/maui#12378**
- Please respond to @vhugogarcia.

---

_From @vhugogarcia on Tuesday, January 3, 2023 3:34:27 PM_

@AllNickNamesAreBusySoITakenThis please refer to this discussion about MAUI community toolkit wanting to implement a map handler on webview. Please join that conversatio, any feedback is always welcome.

**Comment by mattleibow** [comment]
---
**Issue moved from dotnet/maui#12378**
- Please respond to @Ghostbird.

---

_From @Ghostbird on Tuesday, January 3, 2023 4:30:40 PM_

Ah, good idea. This sounds like a better fit for the community toolkit.

EDIT: There's only a discussion about a windows specific control that is intended as a stopgap solution because MAUI doesn't support Windows yet with their map component. It has a [tangential remark](https://github.com/CommunityToolkit/Maui/issues/605#issuecomment-1263653226) that a generic OSM map component would be useful, however I don't completely agree with that. I'd much rather have the option to switch MAUI's map control to a different tile/image source. That might be a community feature rather than a core MAUI feature though.

This would probably involve work on both sides. If MAUI has the correct level of separation of concerns and accessible layers of abstraction, the extra implementations can be part of the community toolkit.

I'm think there must be _some_ separation between:

- Image/Tile source
- Projection
- Rendering
- Control
- Interaction

**Comment by mattleibow** [comment]
---
**Issue moved from dotnet/maui#12378**
- Please respond to @vhugogarcia.

---

_From @vhugogarcia on Tuesday, January 3, 2023 4:45:37 PM_

I agree. @Ghostbird https://github.com/CommunityToolkit/Maui/issues/605 please share the feedback above on the community toolkit thread, so others can see it. 👍🏻

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
